lib/db/meta.go

```diff
@@ -196,8 +200,8 @@ func (m *metadataTracker) removeFile(dev protocol.DeviceID, f FileIntf) {
 	}
 }
 
-func (m *metadataTracker) removeFileLocked(dev protocol.DeviceID, flags uint32, f FileIntf) {
-	cp := m.countsPtr(dev, f.FileLocalFlags())
```

Comment:
I’m not sure about this change. Like, if we have a recv only folder we might add a bunch of files with the local change bit. Later we change the folder type, the files are going to be reprocessed. Should we not remove them with the same bits they were added?

Reply:
Yes and we do by calling `removeFileLocked` with all the bits of `f.FileLocalFlags` separately, see https://github.com/syncthing/syncthing/blob/713ea31f9bf807c8f91d334f8c8758db14ce36a8/lib/db/meta.go#L197. Same when adding files https://github.com/syncthing/syncthing/blob/713ea31f9bf807c8f91d334f8c8758db14ce36a8/lib/db/meta.go#L143, where the locked variant already does use the `flags` argument: https://github.com/syncthing/syncthing/blob/713ea31f9bf807c8f91d334f8c8758db14ce36a8/lib/db/meta.go#L165

Currently `flags` is entirely unused in `removeFileLocked`.